ports/samd/modules/_boot.py

```diff
@@ -1,19 +1,44 @@
 import gc
 import uos
 import samd
+from flashbdev import FlashBdev
+
+flashfs = False
+# Try to mount an external flash drive.
+try:
+    if hasattr(samd, "QSPIflash"):
+        bdev = FlashBdev(samd.QSPIflash())
```

Comment:
I'm not sure why the `FlashBdev` class is needed, it is a very thin wrapper.  Why not just make `QSPIflash`, `SPIflash` and `MCUflash` implement the block protocol?  Then they can be used directly (and the existing `samd.Flash` could just be renamed to `samd.MCUFlash`).

Reply:
Thank you for reviewing. `FlashBdevI` is intentionally as small as possible, and I  wanted to be able to support more than one flash type at a device, like external flash and mcuflash. The block device driver was initially combined with the mcuflash driver.

About `drivers/memory/spiflash.[ch]`. I had seen them sometime during the development. The seemed to me pretty specific for Pyboard, and I could not tell how to use them, besides seeming quite large..